module/zfs/mmp.c

```diff
@@ -206,79 +206,61 @@ typedef enum mmp_vdev_state_flag {
 } mmp_vdev_state_flag_t;
 
 static vdev_t *
-mmp_random_leaf_impl(vdev_t *vd, int *fail_mask)
+get_head_and_rotate(list_t *list)
 {
-	int child_idx;
+	ASSERT(list);
+	vdev_t *leaf;
 
-	if (vd->vdev_ops->vdev_op_leaf) {
-		vdev_t *ret;
+	leaf = list_remove_head(list);
+	if (leaf)
+		list_insert_tail(list, leaf);
```

Comment:
nit: we know this can never be NULL since we've asserted the list isn't empty and we're always reinserting the vdev.  Including the conditional, at least to me, implies this could be NULL, I'd be OK dropping it.  We could always add a comment to this effect.

Reply:
Dropped it